apps/asn1pars.c

```diff
@@ -170,17 +170,17 @@ int asn1parse_main(int argc, char **argv)
     if (derfile && (derout = bio_open_default(derfile, 'w', FORMAT_ASN1)) == NULL)
         goto end;
 
+    if ((buf = BUF_MEM_new()) == NULL)
+        goto end;
     if (strictpem) {
-        if (PEM_read_bio(in, &name, &header, &str, &num) !=
-            1) {
+        if (PEM_read_bio(in, &name, &header, &str, &num) != 1) {
             BIO_printf(bio_err, "Error reading PEM file\n");
             ERR_print_errors(bio_err);
             goto end;
         }
+        buf->data = (char *)str;
+        buf->length = buf->max = num;
```

Comment:
_(Side note: Huh, it seems like `BUF_MEM` was forgotten when the structures were made opaque? A public struct with a rudimentary API. I would have expected to see something like `BUF_MEM_set(buf, data, length)` or `BUF_MEM_attach(buf, data, length)` ..._